### ISSUE TITLE
`List` constructor fails for freshly unserialized elements

### ISSUE THREAD

**Issue by LTLA** [Bioconductor/S4Vectors#103]
```r
library(Matrix)
saveRDS(file="blah.rds", rsparsematrix(10, 10, 0.1))
```

And then, in a new session:

```r
X <- readRDS("blah.rds")
S4Vectors::List(X)
## Error in getClass(element.type) : “dgCMatrix” is not a defined class
```

It's fine if I do anything with `X` before attempting construction; it seems the S4 machinery kicks in to attach **Matrix**:

```r
X <- readRDS("blah.rds")
dim(X)
S4Vectors::List(X) # ok
## List of length 1
```

This is most relevant when dealing with parallel code where a class-defining package was loaded in the workers but is not available in the master thread, thus causing errors when I try to coerce the `bplapply` output into a `List`.

<details>
<summary>Session information</summary>

```
R Under development (unstable) (2022-02-11 r81718)
Platform: x86_64-apple-darwin19.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Users/luna/Software/R/trunk/lib/libRblas.dylib
LAPACK: /Users/luna/Software/R/trunk/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods
[8] base

other attached packages:
[1] S4Vectors_0.33.15   BiocGenerics_0.41.2 Matrix_1.4-1

loaded via a namespace (and not attached):
[1] compiler_4.2.0  grid_4.2.0      lattice_0.20-45
```
</details>

**Comment by hpages** [comment]
Fixed in **S4Vectors** 0.33.16 (see 38abc42ca7f5f0bb7a4fe1c7561fc9d8f7d170c5).

**Comment by hpages** [comment]
Oh, but that isn't good either:
```
X <- readRDS("blah.rds")

## First dim() call returns NULL:
# dim(X)
# Loading required package: Matrix
# NULL

## Only 2nd dim() call actually returns the dimensions:
dim(X)
# [1] 10 10
```
@lawremi Maybe something for the **methods** package?

**sessionInfo():**
```
R Under development (unstable) (2022-03-24 r81969)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 21.10

Matrix products: default
BLAS:   /home/hpages/R/R-4.2.r81969/lib/libRblas.so
LAPACK: /home/hpages/R/R-4.2.r81969/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB              LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Matrix_1.4-1

loaded via a namespace (and not attached):
[1] compiler_4.2.0  grid_4.2.0      lattice_0.20-45
```
